### ISSUE TITLE
Build new Testimonial component

### ISSUE THREAD

**Issue by kwm14** [vanilla-framework/vanilla-framework#1961]
A pattern in Vanilla needs to be flexible enough that it can be used across a wide range of scenarios and shouldn't be constrained to a specific number of applications.

- Original design issue: https://github.com/ubuntudesign/vanilla-design/issues/211
- Design spec: https://github.com/ubuntudesign/vanilla-design/tree/master/Testimonial

## Visual
![basic-testimonial](https://user-images.githubusercontent.com/17748020/44219880-02802c00-a175-11e8-83d0-66bdc37d91e6.png)

## Context

Currently we have a few variations of this component so we needed to align and have a consistent style to use across our suite of sites.

**Comment by deadlight** [comment]
@kwm14 

Originally, the testimonial designs were quite radically different to quotes but now this pattern is a duplicate of the pull quote pattern (text in various sizes with quotes and a citation) and I don't think that we need it. In fact, I think it'll lead to confusing inconsistencies. 

How about I add the logo option to the existing pull quote.

**Comment by deadlight** [comment]
To illustrate, since this issue was created we've added the small quote style that seems fitting for this:

![image](https://user-images.githubusercontent.com/118614/53870423-1b0d4200-3ff2-11e9-840e-6fa675e13861.png)

![image](https://user-images.githubusercontent.com/118614/53870447-295b5e00-3ff2-11e9-950d-c8d2a6cfdd17.png)

**Comment by kwm14** [comment]
@deadlight I forgot to add header/image version of the testimonial to the visual. I've added it now above 😄

**Comment by deadlight** [comment]
@kwm14 yeah, I found that. My proposal is to add the header/image part to the existing pullquote styling to save creating an extra component.

**Comment by kwm14** [comment]
@deadlight yeah sounds good 👍

**Comment by deadlight** [comment]
In the docs, we call them "quotes" rather than pull-quotes, which seems like a better, more generic name (which can work for a testimonial or a pull quote)
